internal/service/storage/storage_backup.go

```diff
@@ -0,0 +1,234 @@
+	data.ID = types.StringValue(backupDetails.UUID)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *storageBackupResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data storageBackupModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.ID.ValueString() == "" {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Get backup details
+	backupDetails, err := r.client.GetStorageDetails(ctx, &request.GetStorageDetailsRequest{
+		UUID: data.ID.ValueString(),
+	})
+	if err != nil {
+		if utils.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to read backup details",
+				utils.ErrorDiagnosticDetail(err),
+			)
+		}
+		return
+	}
+
+	data.ID = types.StringValue(backupDetails.UUID)
+	data.CreatedAt = types.StringValue(backupDetails.Created.Format(time.RFC3339))
+	data.Title = types.StringValue(backupDetails.Title)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *storageBackupResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read current configuration
+	var data storageBackupModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read the current state
+	var state storageBackupModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateRequired := false
+	modifyStorageRequest := request.ModifyStorageRequest{
+		UUID: state.ID.ValueString(),
+	}
+
+	if data.Title.ValueString() != state.Title.ValueString() {
+		modifyStorageRequest.Title = data.Title.ValueString()
+		updateRequired = true
+	}
+
+	if updateRequired {
+		_, err := r.client.ModifyStorage(ctx, &modifyStorageRequest)
+		if err != nil {
+			resp.Diagnostics.AddError("Failed to update storage backup", utils.ErrorDiagnosticDetail(err))
+			return
+		}
+	}
+
+	data.ID = state.ID
+	data.CreatedAt = state.CreatedAt
+	data.SourceStorage = state.SourceStorage
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *storageBackupResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state storageBackupModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	backupID := state.ID.ValueString()
+
+	// Wait for backup to enter 'online' state as storage devices can only be deleted in this state.
+	_, err := r.client.WaitForStorageState(ctx, &request.WaitForStorageStateRequest{
+		UUID:         backupID,
+		DesiredState: upcloud.StorageStateOnline,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Storage did not reach online state",
+			utils.ErrorDiagnosticDetail(err),
+		)
+		return
+	}
+
+	// Do we want to delete the snapshot from the system or only make TF think it is deleted?
```

Comment:
I guess expected behavior would be to delete the backup. If we only remove it from Terraform, customer will continue to pay for the backup even though they think it was deleted 🤔 

```suggestion
```